test/Streamly/Test/Prelude/Serial.hs

```diff
@@ -218,9 +258,9 @@ splitterProperties sep desc = do
         where
 
         testCase xs = do
-            ys <- splitter xs (replicate i sep)
+            ys <- liftIO $ splitter xs (replicate i sep)
             szs <-
-                IS.toList
+                liftIO $ toList
```

Comment:
Why not use `liftIO` in toList/Splitter etc or just make their signature polymorphic.